File: sentimental-cash/cash.py
from cs50 import get_float

quarters = 0
dimes = 0
nickel = 0
pennies = 0


while True:
    try:
        user_input = get_float("Change: ")
        cents = round(user_input * 100)
        if user_input > 0:
            break
    except ValueError:
        print("")

while cents > 0:
    if cents >= 25:
        cents = cents - (25)
        quarters = quarters + 1

    elif cents >= 10:
        cents = cents - (10)
        dimes = dimes + 1

    elif cents >= 5:
        cents = cents - (5)
        nickel = nickel + 1

    elif cents >= 1:
        cents = cents - (1)
        pennies = pennies + 1

print(quarters + dimes + nickel + pennies)

File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Prompt the user for change owed, in cents
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    // Calculate how many quarters you should give customer
    // Subtract the value of those quarters from cents

    int quarters = calculate_quarters(cents);
    cents = cents - (quarters * 25);

    int dimes = calculate_dimes(cents);
    cents = cents - (dimes * 10);

    int nickels = calculate_nickels(cents);
    cents = cents - (nickels * 5);

    int pennies = calculate_pennies(cents);
    cents = cents - (pennies * 1);

    int total_coins = quarters + dimes + nickels + pennies;
    printf("%i\n", total_coins);
}
int calculate_quarters(int cents)
{
    // Calculate how many quarters you should give customer
    int quarters = 0;
    while (cents >= 25)
    {
        quarters++;
        cents = cents - 25;
    }
    return quarters;
}

int calculate_dimes(int cents)
{
    // Calculate how many dimes you should give customer
    int dimes = 0;
    while (cents >= 10)
    {
        dimes++;
        cents = cents - 10;
    }
    return dimes;
}

int calculate_nickels(int cents)
{
    // Calculate how many nickels you should give customer
    int nickels = 0;
    while (cents >= 5)
    {
        nickels++;
        cents = cents - 5;
    }
    return nickels;
}

int calculate_pennies(int cents)
{
    // Calculate how many pennies you should give customer
    int pennies = 0;
    while (cents >= 1)
    {
        pennies++;
        cents = cents - 1;
    }
    return pennies;
}
